modules/pinger/src/main/java/org/hawkular/component/pinger/PingManager.java

```diff
@@ -16,114 +16,189 @@
+        org.hawkular.inventory.api.Environments.Single env = null;
+        try {
+            env = tenant.environments().create(
+            Environment.Blueprint.builder().withId(envId).build());
+        } catch (EntityAlreadyExistsException e) {
+            env = tenant.environments().get(envId);
+        }
+        env.feedlessResources().create(
+                Resource.Blueprint.builder().withId(UUID.randomUUID().toString()).withResourceType("URL")
+                        .withProperty("url", url).build());
+
+    }
 
     @PostConstruct
     public void startUp() {
-        int attempts = 0;
-        while (attempts++ < 10) {
-            try {
-                Client client = ClientBuilder.newClient();
-                final String inventoryUrl = configuration.getInventoryBaseUri();
-                WebTarget target = client.target(inventoryUrl +"/"+ tenantId + "/resourceTypes/URL/resources");
-                Response response = target.request().get();
-
-                if (isResponseOk(response.getStatus())) {
-                    List<?> list = response.readEntity(List.class);
-                    if (list.isEmpty()) {
-                    } else {
-                        for (Object o : list) {
-                            if (o instanceof Map) {
-                                Map<?, ?> m = (Map<?, ?>) o;
-                                String id = (String) m.get("id");
-                                @SuppressWarnings("unchecked")
-                                Map<String, String> params = (Map<String, String>) m.get("properties");
-                                String url = params.get("url");
-                                String method = params.get("method");
-                                destinations.add(new PingDestination(id, url, method));
-                            }
-                        }
-                    }
-                    response.close();
-                    client.close();
-                    return;
-                } else {
-                    Log.LOG.wNoInventoryFound(response.getStatus(), response.getStatusInfo().getReasonPhrase());
-                }
-            } catch (Exception e) {
-                Log.LOG.wNoInventoryFound(-1, "Exception while trying to reach or read response of inventory: "
-                        + e.getMessage());
-            }
 
-            try {
-                Thread.sleep(2000);
-            } catch (InterruptedException e) {
-                Thread.currentThread().interrupt();
-                return;
-            }
+        /*
+         * Add the observer before reading the existing URLs from the inventory so that we do not loose the URLs that
+         * could have been added between those two calls.
+         */
+        inventory.observable(Interest.in(Resource.class).being(Action.created())).subscribe(newUrlsCollector);
+
+        Set<Resource> urls = inventory.tenants().getAll().resourceTypes().getAll(With.id(PingDestination.URL_TYPE))
```

Comment:
Yes, pinger is now required to run co-located with Hawkular. Doing it "properly" and enable remote pinger would be too much work for the next iteration. This will have to be revisited of course, but I think this is actually kinda good for the SaaS usecase, because using this approach we can circumvent the authz that is now in place on the REST endpoint. Being able to authz the pinger requires much more work on the pinger side as well as inventory side to do it "properly" (it could be made working with inventory as it is now, but in a little bit hacky way).